### PR TITLE
feat: Blockquote component

### LINKED ISSUE #5820: Create `Blockquote` Component
Create a `Blockquote` component using the new [figma design (direct link)](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?type=design&node-id=1509-16204&mode=dev). 

> [Read more about the Node.js Website redesign](https://github.com/nodejs/nodejs.org/discussions/5131)

---

### Before You Start...

- Use the [Contributor's Guide](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) to learn how to [setup your environment](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#getting-started), run [available commands](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#cli-commands), and construct [proper commits](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#commit-guidelines).
- Review the [technologies used](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#technologies-used-in-the-website) when building the site, and [why they were chosen](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#remarks-on-technologies-used).
- Review the [component creation guide](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#creating-components) to follow our standards.

---

### Details

image

❓ [Not sure where to start breaking down the figma? Read this short guide](https://github.com/nodejs/nodejs.org/discussions/5131#discussioncomment-6848378)

- Create a new directory called `Blockquote` within the [`Common` directory](https://github.com/nodejs/nodejs.org/tree/main/components/Common) that contains all the new code.
- Use the new tailwind color and typography system to assign properties. For example, `Light` uses `Green/600` per the figma.
- Add a storybook story file `components/Common/Blockquote/index.stories.tsx` which exercises each of the button's states.
- Add the em-dash `—` for users before the `attribution` prop renders

Suggested props include:

- `children` text content, rendered within a `p` tag: `“In a world older and more complete than ours...` in the figma
- `attribution` (string) : `Hayden Bleasel, Stoic Philosopher` in the figma

There are 2 states to capture within styles and stories:

- Light
- Dark

### Open Questions, to Discuss

- how should the semantic markup be structured? this is a topic of much debate, it seems. suggest starting with [MDN's approach](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/blockquote)
- how does a user-supplied markdown blockquote `> text` render in tailwind? that will certainly be something that comes up. I suggest we add a storybook story also rendering this (but not using the `Blockquote` component, as markdown-driven content won't have `attribution` in many cases. I don't think an MDX component map is appropriate right now either.

### PR DESCRIPTION
**Pull request by canerakdas**
## Description

This pull request addresses the issue(#5820)  by adding a new directory called Blockquote within the Common directory. This new directory contains all the code necessary for the Blockquote component.

### Changes

- Created a new directory: `Common/Blockquote`.
- Implemented the Blockquote component using the new Tailwind color, typography system, and Figma design.
- Added a Storybook story file: `components/Common/Blockquote/index.stories.tsx`, which exercises each of the Blockquote component's states.
	- Default
	- Without Attribution
	- Nested

## Validation

image
image

## Related Issues

fixes #5820 

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: components/Common/Blockquote/index.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+} & ComponentProps<'blockquote'>;
+
+const Blockquote: FC<PropsWithChildren<BlockquoteProps>> = ({
+  children,
+  cite,
+  attribution,
+}) => (
+  <blockquote className={styles.wrapper} cite={cite}>
+    {children}
+    {attribution && (
```
Comment (ovflowd) [inline components/Common/Blockquote/index.tsx]: I don't recall attribution being part of the blockquote component spec. I don't even think MDX can actually map that.

**Comment by ovflowd** [comment]
@canerakdas the code looks good, but can you rebase the PR? I just want to check the light/dark theme switching properly with the new stuff we added.

**Comment by ovflowd** [comment]
Also note that we're switching to PostCSS so you might need to rename your `.scss` file to `.css` once https://github.com/nodejs/nodejs.org/pull/5888 gets merged.

**Comment by canerakdas** [comment]
> Also note that we're switching to PostCSS so you might need to rename your `.scss` file to `.css` once #5888 gets merged.

Sure, I'm tracking 👍

**Comment by canerakdas** [comment]
While rebasing, I should have merged onto a different branch. Instead of fixing that, I will open a new PR

**Pull request closed without merging**